### ISSUE TITLE
Unexpected Address parsing error 

### ISSUE THREAD

**Issue by damip** [massalabs/massa-as-sdk#64]
Students showed me this in a smart contract: 

![20221120_190402](https://user-images.githubusercontent.com/10211103/203016139-ff526371-4224-4e0b-9416-0a28fdfc147e.jpg)

When the lines containing "Balance before" and "Balance after" were present, they had a runtime error saying "addr parsing error". When they commented those two lines, the problem disappeared.

**Comment by gregLibert** [comment]
This error is really weird: the only reference I found about a balance() function is in massa-as-sdk (balance wrapping the ABI call balance).
I also looked at the ERC20 wrapper but the function is not named like that and there is no address manipulated there.

Nonetheless, I looked at the Amount package and found some issues there and some code not updated.
Hence this [issue](https://github.com/massalabs/as/issues/36) but I'm pretty sure it's not the root cause...
